packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -286,7 +286,11 @@ contract Bridge is EssentialContract, IBridge {
                     // - need to have a buffer/small revenue to the realyer since it consumes
                     // maintenance and infra costs to operate
                     uint256 refund = stats.numCacheOps * _GAS_REFUND_PER_CACHE_OPERATION;
-                    stats.gasUsedInFeeCalc = uint32(GAS_OVERHEAD + gasStart - gasleft());
+                    // Taking into account the encoded message calldata cost, and can count with 16
+                    // gas per bytes (vs. checking each and every byte if zero or non-zero)
+                    uint32 msgCalldataGas = uint32(abi.encode(_message).length) << 4;
```

Comment:
the message only have the data file as a variable length data, all the other fields are fixed size. Maybe we can use _message.data.lenght + a constant value to avoid `abi.encode`?

Reply:
Done. Needed to get rid of this tmp variable tho, due to stack too deep.